src/main/scala/mesosphere/marathon/core/launchqueue/impl/LaunchQueueDelegate.scala

```diff
@@ -31,8 +31,10 @@ private[launchqueue] class LaunchQueueDelegate(
   override def notifyOfInstanceUpdate(update: InstanceChange): Future[Done] =
     askQueueActorFuture[InstanceChange, Done]("notifyOfInstanceUpdate")(update)
 
-  override def add(runSpec: RunSpec, count: Int): Future[Done] =
+  override def add(runSpec: RunSpec, count: Int): Future[Done] = {
+    require(runSpec.role.isDefined, "Adding a runSpec to launch queue requires a valid role in runSpec")
```

Comment:
Why make `runSpec.role` optional? I think it should be always defined so why not express it in the type system?

Reply:
Non-Optional now